server/retrieval/cache.go

```diff
@@ -0,0 +1,43 @@
+package retrieval
+
+import (
+	"context"
+	"fmt"
+
+	lru "github.com/hashicorp/golang-lru/v2"
```

Comment:
Just thinking aloud here: I wonder if their [ARC cache](https://github.com/hashicorp/golang-lru/blob/main/arc/arc.go) would perform better - I imagine delegations higher up the chain are accessed fairly frequently? but then get evicted quickly as lower parts of the chain are cached... 🤷 not sure though.